api/internal/plugins/loader/loader.go

```diff
@@ -53,16 +53,21 @@ func (l *Loader) SetWorkDir(wd string) {
 }
 
 func (l *Loader) LoadGenerators(
-	ldr ifc.Loader, v ifc.Validator, rm resmap.ResMap) ([]resmap.Generator, error) {
-	var result []resmap.Generator
+	ldr ifc.Loader, v ifc.Validator, rm resmap.ResMap) (
+	result []resmap.Generator, origins []*resource.Origin, err error) {
 	for _, res := range rm.Resources() {
 		g, err := l.LoadGenerator(ldr, v, res)
 		if err != nil {
-			return nil, err
+			return nil, nil, err
+		}
+		generatorOrigin, err := resource.OriginFromCustomPlugin(res)
+		if err != nil {
+			return nil, nil, err
 		}
 		result = append(result, g)
+		origins = append(origins, generatorOrigin)
```

Comment:
Adding an `Origin()` method to the generator interface is a really big code change that will require all generators to implement this method - probably not what we want - but I've created a wrapper for generator, `GeneratorWithProperties`, so that we can add an `Origin` attribute to store a generator and its corresponding origin.

Reply:
> Adding an Origin() method to the generator interface is a really big code change that will require all generators to implement this method

🤦‍♀️ of course we don't want to do that. I really like the wrapper solution.